pkg/server/plugin/bundlepublisher/awss3/awss3_test.go

```diff
@@ -0,0 +1,418 @@
+		},
+		{
+			name: "client error",
+			config: &Config{
+				AccessKeyID:     "access-key-id",
+				SecretAccessKey: "secret-access-key",
+				Region:          "region",
+				Bucket:          "bucket",
+				ObjectKey:       "object-key",
+				Format:          "spiffe",
+			},
+			expectCode:   codes.Internal,
+			expectMsg:    "failed to create client: client creation error",
+			newClientErr: errors.New("client creation error"),
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			var err error
+			options := []plugintest.Option{
+				plugintest.CaptureConfigureError(&err),
+				plugintest.CoreConfig(catalog.CoreConfig{
+					TrustDomain: spiffeid.RequireTrustDomainFromString("example.org"),
+				}),
+				plugintest.ConfigureJSON(tt.config),
+			}
+
+			newClient := func(awsConfig aws.Config) (simpleStorageService, error) {
+				if tt.newClientErr != nil {
+					return nil, tt.newClientErr
+				}
+				return &fakeClient{
+					awsConfig: awsConfig,
+				}, nil
+			}
+			p := newPlugin(newClient)
+
+			plugintest.Load(t, builtin(p), nil, options...)
+			spiretest.RequireGRPCStatusHasPrefix(t, err, tt.expectCode, tt.expectMsg)
+
+			if tt.expectMsg != "" {
+				require.Nil(t, p.config)
+				return
+			}
+
+			// Check that the plugin has the expected configuration.
+			tt.config.bundleFormat, err = bundleformat.FromString(tt.config.Format)
+			require.NoError(t, err)
+			require.Equal(t, tt.config, p.config)
+
+			client, ok := p.s3Client.(*fakeClient)
+			require.True(t, ok)
+
+			// It's important to check that the configuration has been wired
+			// up to the aws config, that needs to have the specified region
+			// and credentials.
+			require.Equal(t, tt.config.Region, client.awsConfig.Region)
+			creds, err := client.awsConfig.Credentials.Retrieve(context.Background())
+			require.NoError(t, err)
+			require.Equal(t, tt.config.AccessKeyID, creds.AccessKeyID)
+			require.Equal(t, tt.config.SecretAccessKey, creds.SecretAccessKey)
+		})
+	}
+}
+
+func TestPublishBundle(t *testing.T) {
+	testBundle := getTestBundle(t)
+
+	for _, tt := range []struct {
+		name string
+
+		newClientErr     error
+		expectCode       codes.Code
+		expectMsg        string
+		config           *Config
+		bundle           *types.Bundle
+		putObjectErr     error
+		testMultiplePuts bool
+	}{
+		{
+			name:   "success",
+			bundle: testBundle,
+			config: &Config{
+				AccessKeyID:     "access-key-id",
+				SecretAccessKey: "secret-access-key",
+				Region:          "region",
+				Bucket:          "bucket",
+				ObjectKey:       "object-key",
+				Format:          "spiffe",
+			},
+		},
+		{
+			name:   "multiple times",
+			bundle: testBundle,
+			config: &Config{
+				AccessKeyID:     "access-key-id",
+				SecretAccessKey: "secret-access-key",
+				Region:          "region",
+				Bucket:          "bucket",
+				ObjectKey:       "object-key",
+				Format:          "spiffe",
+			},
+			testMultiplePuts: true,
+		},
```

Comment:
looks like this test case must be removed? since testMutiplePuts is never used

Reply:
Forgot to remove it, thanks!